File: src/015/p015.cpp
#include <iostream>
using namespace std;

long combination(long n, long k) {
    long c = 1;
    for (long i = 0; i < k; i++) {
        c *= n - i;
        c /= i + 1;
    }
    return c;
}

long solve() {
    return combination(40, 20);
}

int main() {
    long result = solve();
    cout << result << endl;
}

File: src/015/p015.lua
function combination(n, k)
  local c = 1
  for i=0, k-1 do
    c = c * (n - i)
    c = c / (i + 1)
  end
  return math.floor(c)
end

function solve()
  return combination(40, 20)
end

local result = solve()
print(result)
